tests/unit/lib/queuePopulator/kafkaLogConsumer/LogConsumer.js

```diff
@@ -0,0 +1,219 @@
+                            type: 'put',
+                            value: {
+                                field: 'value'
+                            },
+                        }],
+                    });
+                    return done();
+                });
+            });
+        });
+    });
+
+    describe('readRecords', () => {
+        it('Should return stream', done => {
+            const waitAssignementStub = sinon.stub(logConsumer, '_waitForAssignment')
+                .callsArg(1);
+            const storeOffsetsStub = sinon.stub(logConsumer, '_storeCurrentOffsets')
+                .callsArg(0);
+            const consumeKafkaStub = sinon.stub(logConsumer, '_consumeKafkaMessages')
+                .callsArg(1);
+            logConsumer._resetRecordStream();
+            logConsumer.readRecords({ limit: 1 }, (err, res) => {
+                assert(waitAssignementStub.called);
+                assert(storeOffsetsStub.called);
+                assert(consumeKafkaStub.called);
+                assert.ifError(err);
+                assert(res.log instanceof ListRecordStream);
+                assert.strictEqual(res.tailable, false);
+                assert.strictEqual(typeof(res.log.getOffset), 'function');
+                return done();
+            });
+        });
+
+        it('Should fail if consumer group failed to stabilize', done => {
+            const waitAssignementStub = sinon.stub(logConsumer, '_waitForAssignment')
+                .callsArgWith(1, errors.InternalError);
+            logConsumer._resetRecordStream();
+            logConsumer.readRecords({ limit: 1 }, err => {
+                assert(waitAssignementStub.called);
+                assert.deepEqual(err, errors.InternalError);
+                return done();
+            });
+        });
+
+        it('Should fail if it can\'t store offsets', done => {
+            const waitAssignementStub = sinon.stub(logConsumer, '_waitForAssignment')
+                .callsArg(1);
+            const storeOffsetsStub = sinon.stub(logConsumer, '_storeCurrentOffsets')
+                .callsArgWith(0, errors.InternalError);
+            logConsumer._resetRecordStream();
+            logConsumer.readRecords({ limit: 1 }, err => {
+                assert(waitAssignementStub.called);
+                assert(storeOffsetsStub.called);
+                assert.deepEqual(err, errors.InternalError);
+                return done();
+            });
+        });
+
+        it('Should fail if it can\'t consume kafka messages', done => {
+            const waitAssignementStub = sinon.stub(logConsumer, '_waitForAssignment')
+                .callsArg(1);
+            const storeOffsetsStub = sinon.stub(logConsumer, '_storeCurrentOffsets')
+                .callsArg(0);
+            const consumeKafkaStub = sinon.stub(logConsumer, '_consumeKafkaMessages')
+                .callsArgWith(1, errors.InternalError);
+            logConsumer._resetRecordStream();
```

Comment:
do we need to call reset with each test, isn't log consumer re-instantiated?

Reply:
I think the reason i did that was because i stubbed the `_consumeKafkaMessages` function, and that's the only place where we do stream initialization.
But looking at it again, we don't even need the stream in some of those test cases as we throw an error before reaching the code that uses the stream in the `readRecords` function